src/test/java/ua/lviv/iot/lab2/manager/impl/DwellingManagerTest.java

```diff
@@ -1,97 +1,117 @@
 package ua.lviv.iot.lab2.manager.impl;
 
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.stream.IntStream;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import org.junit.jupiter.api.*;
 import ua.lviv.iot.lab2.models.*;
-import java.util.*;
-
-import static org.junit.jupiter.api.Assertions.*;
 
 public class DwellingManagerTest {
-    private final DwellingManager manager = new DwellingManager();
-    List<Dwelling> dwellings = generateListOfDwellings();
+    DwellingManager manager;
+    List<Dwelling> dwellings;
+
     @BeforeAll
     static void setUpBeforeClass() throws Exception {
     }
+
     @AfterAll
-    static void tearDownAfterClass() throws Exception{
+    static void tearDownAfterClass() throws Exception {
+
     }
+
     @BeforeEach
     void setUp() throws Exception {
+        manager = new DwellingManager();
+        dwellings = generateListOfDwellings();
         manager.addDwelling(dwellings);
     }
+
     @AfterEach
-    void tearDown() throws Exception{
+    void tearDown() throws Exception {
     }
+
     @Test
-    public void testGeneral(){
+    public void testGeneral() {
         manager.sortByPrice(dwellings, Choice.ASCENDING);
         assertNotNull(manager.toString());
         manager.sortByPrice(dwellings, Choice.DESCENDING);
         assertNotNull(manager.toString());
-        assertNotNull(manager.findByName(dwellings,"Provesin"));
-        assertNotNull(manager.findByLocation(dwellings,"9, Stusa"));
+        assertNotNull(manager.findByName(dwellings, "Provesin"));
+        assertNotNull(manager.findByLocation(dwellings, "9, Stusa"));
     }
 
     @Test
     void addDwellingTest() {
-        Assertions.assertFalse(manager.getAllDwellings().isEmpty());
-        Assertions.assertEquals(3,manager.getAllDwellings().size());
+        Assertions.assertFalse(manager.getDwellingsMap().isEmpty());
+        Assertions.assertEquals(4, manager.getDwellingsMap().size());
     }
 
     @Test
     void findByNameTest() {
-        assertEquals(" Name='Provesin', location='162, Glynyansky Trakt St."
-                ,manager.findByName(dwellings,"Provesin").toString());
+        var expectedDwellingName = "Provesin";
```

Comment:
If you have an opportunity to give a variable more clear type, I think you should